### PR TITLE
Guestbook: Addition troubleshooting section.

### PR DESCRIPTION
**Pull request by jayunit100**
Here's the troubleshooting section from the mailing list threads https://groups.google.com/forum/#!topic/google-containers/nr92wX4p3VY !

### CODE REVIEW

File: examples/guestbook/README.md [hunk 1]
```diff
@@ -383,3 +383,60 @@ kubectl delete -f examples/guestbook/redis-slave-service.json
 kubectl delete pod redis-master # This is the only pod that requires manual removal.
 ```
 
+### Troubleshooting
+
+Guestbook can go wrong for a variety of reasons, which makes it an effective test.  Lets test the web app simply using *curl*, so we can see whats going on.
```
Comment (davidopp) [inline examples/guestbook/README.md]: Maybe say "the Guestbook example can fail for a variety of reasons..." ?

Also, maybe mention the setup you were using when you ran into these problems, so people can judge how near or far they are from your setup.

File: examples/guestbook/README.md [hunk 2]
```diff
@@ -383,3 +383,60 @@ kubectl delete -f examples/guestbook/redis-slave-service.json
...
+```
+
+1) When the *Web page hasn't come up yet*:
+
+When you go to localhost:8000, you might not see the page at all.  Testing it with curl...
+```shell
+   ==> default: curl: (56) Recv failure: Connection reset by peer
+```
+
+This means the web frontend isn't up yet.  Wait a while, possibly about 2 minutes or more, depending on your set up. Also, run a *watch* on docker ps, to see if containers are cycling on and off or not starting.
```
Comment (davidopp) [inline examples/guestbook/README.md]: Add instructions on how to run watch on docker ps?

File: examples/guestbook/README.md [hunk 3]
```diff
@@ -383,3 +383,60 @@ kubectl delete -f examples/guestbook/redis-slave-service.json
...
+
+2) *When Redis can't connect to master*:
+
+```shell
+<b>Fatal error</b>:  Uncaught exception 'Predis\Connection\ConnectionException' with message 'php_network_getaddresses: getaddrinfo failed: Name or service not known [tcp://:0]' in /vendor/predis/predis/lib/Predis/Connection/AbstractConnection.php:141
+```
+
+The fix: Make sure that environmental variables are being set correctly.
```
Comment (davidopp) [inline examples/guestbook/README.md]: Give an example of which env vars you're talking about? (maybe the ones that weren't being set when you ran this?)

File: examples/guestbook/README.md (hunk 2 above)
Comment (jayunit100) [inline examples/guestbook/README.md]: added below...

**Comment by jayunit100** [comment]
thanks, updated the patch !

File: examples/guestbook/README.md [hunk 4]
```diff
@@ -383,3 +383,74 @@ kubectl delete -f examples/guestbook/redis-slave-service.json
 kubectl delete pod redis-master # This is the only pod that requires manual removal.
 ```
 
+### Troubleshooting
+
+the Guestbook example can fail for a variety of reasons, which makes it an effective test.  Lets test the web app simply using *curl*, so we can see whats going on.
+
+Before we proceed, what are some setup idiosyncracies that might cause the app to fail (or, appear to fail, when merely you have a *cold start* issue.
```
Comment (davidopp) [inline examples/guestbook/README.md]: missing closing paren after "fail"

File: examples/guestbook/README.md [hunk 5]
```diff
@@ -383,3 +383,74 @@ kubectl delete -f examples/guestbook/redis-slave-service.json
 kubectl delete pod redis-master # This is the only pod that requires manual removal.
 ```
 
+### Troubleshooting
+
+the Guestbook example can fail for a variety of reasons, which makes it an effective test.  Lets test the web app simply using *curl*, so we can see whats going on.
```
Comment (davidopp) [inline examples/guestbook/README.md]: the -> The
Lets -> Let's

**Comment by davidopp** [comment]
LGTM

**Pull request merged**